src/core/Akka.Streams/Dsl/RestartFlow.cs

```diff
@@ -481,31 +187,42 @@ internal sealed class RestartWithBackoffFlow<TIn, TOut, TMat> : GraphStage<FlowS
 
         public override FlowShape<TIn, TOut> Shape { get; }
 
-        protected override GraphStageLogic CreateLogic(Attributes inheritedAttributes) => new Logic(this, "Flow");
+        protected override GraphStageLogic CreateLogic(Attributes inheritedAttributes) => new Logic(this, inheritedAttributes, "Flow");
 
         private sealed class Logic : RestartWithBackoffLogic<FlowShape<TIn, TOut>, TIn, TOut>
         {
             private readonly RestartWithBackoffFlow<TIn, TOut, TMat> _stage;
+            private readonly Attributes _inheritedAttributes;
             private Tuple<SubSourceOutlet<TIn>, SubSinkInlet<TOut>> _activeOutIn;
-
-            public Logic(RestartWithBackoffFlow<TIn, TOut, TMat> stage, string name)
+            private TimeSpan _delay;
+            
+            public Logic(RestartWithBackoffFlow<TIn, TOut, TMat> stage, Attributes inheritedAttributes, string name)
                 : base(name, stage.Shape, stage.In, stage.Out, stage.Settings, stage.OnlyOnFailures)
             {
+                _delay = _inheritedAttributes.GetAttribute<Delay>(new Delay(TimeSpan.FromMilliseconds(50))).Duration;
                 _stage = stage;
+                _inheritedAttributes = inheritedAttributes;
                 Backoff();
             }
 
             protected override void StartGraph()
             {
                 var sourceOut = CreateSubOutlet(_stage.In);
                 var sinkIn = CreateSubInlet(_stage.Out);
-                Source.FromGraph(sourceOut.Source).Via(_stage.FlowFactory()).RunWith(sinkIn.Sink, SubFusingMaterializer);
+                
+                var graph = Source.FromGraph(sourceOut.Source)
+                    //temp fix becaues the proper fix would be to have a concept of cause of cancellation. See https://github.com/akka/akka/pull/23909
```

Comment:
we should file a separate issue for this once it's merged, so we can revisit in the future.

Reply:
Agreed. Ill make a note for myself